autoware_plugin/src/autoware_plugin.cpp

```diff
@@ -218,4 +240,171 @@ namespace autoware_plugin
+
+            for(int i = 1; i < original_tp.trajectory_points.size() - 1; i++ )
+            {            
+                for (lanelet::ConstLanelet l : shortest_path) {
+
+                    if (l.id()== std::stoi(original_tp.trajectory_points[i].lane_id)) {
+                        start_lanelet = l;
+                    }
+                }
+
+                cav_msgs::TrajectoryPlanPoint new_tpp;
+                new_tpp.target_time = i * tp / original_tp.trajectory_points.size();
+
+                double down_track_ = polynomial_calc(values,new_tpp.target_time);
+
+                new_tpp.lane_id = original_tp.trajectory_points[i].lane_id;
+                new_tpp.controller_plugin_name = original_tp.trajectory_points[i].controller_plugin_name;
+                new_tpp.planner_plugin_name = original_tp.trajectory_points[i].planner_plugin_name;
+
+                for (auto centerline_point:start_lanelet.centerline2d()) {
+                    double dt = wm_->routeTrackPos(centerline_point).downtrack;
+                    if (dt - down_track_ <= min_downtrack){
+                        new_tpp.x = centerline_point.x();
+                        new_tpp.y = centerline_point.y();
+                        break;
+                    }
+                }
+
+                double x_original = original_tp.trajectory_points[i].x - original_tp.trajectory_points[i - 1].x;
+                double t_original = original_tp.trajectory_points[i].target_time - original_tp.trajectory_points[i - 1].target_time;
+
+                double x_new = new_trajectory_points[i].x - new_trajectory_points[i - 1].x;
+                double t_new = new_trajectory_points[i].target_time - new_trajectory_points[i - 1].target_time;
+
+                double v_new = x_new/t_new;
+                double v_original = x_original/t_original;
+
+                if(v_new > v_original){
+                    carma_wm::TrackPos track_pose = wm_->routeTrackPos(lanelet::BasicPoint2d(original_tp.trajectory_points[i].x, original_tp.trajectory_points[i].y));
+                    values = quintic_coefficient_calculator::quintic_coefficient_calculator(track_pose.downtrack, 
+                                                                                            xt, 
+                                                                                            current_speed_, 
+                                                                                            vt, 
+                                                                                            a0, 
+                                                                                            at, 
+                                                                                            t0, 
+                                                                                            tp);
+                }
+
+                new_tpp.x = polynomial_calc(values,new_tpp.target_time);
```

Comment:
this doesnt make sence. If you need to update trajectory you need to do it with downtrack like above

Reply:
I think you just add a TODO for updating the trajectory properly.